csharp/src/Google.Protobuf/MessageParser.cs

```diff
@@ -287,6 +302,19 @@ public new T ParseFrom(Stream input)
             return message;
         }
 
+        /// <summary>
+        /// Parses a message from the given sequence.
+        /// </summary>
+        /// <param name="data">The data to parse.</param>
+        /// <returns>The parsed message.</returns>
+        [SecuritySafeCritical]
+        public new T ParseFrom(ReadOnlySequence<byte> data)
```

Comment:
Only used in benchmarks

Reply:
I improved the GeneratedMessageTest so that it verifies multiple codepaths (I borrowed the MessageParsingHelpers from your original PR which works very well for that purpose) and we should have some basic coverage now.